src/metaopt/io/database/mongodb.py

```diff
@@ -88,26 +90,16 @@ def unlock(self):
             raise_from(DatabaseError("Could not determine unlock due to connection."), e)
         return unlock
 
-    def insert(self, collection_name, document_dictionaries):
-        """ Inserts a document/multiple documents
-
-        Keyword arguments:
-        collection_name -- the collection name
-        document_dictionaries -- the dictionary/dictionaries that would be inserted
-                                 inside the collection
-        Returns:
-        result -- an instance of InsertManyResult which has inserted_ids and acknowledged
-        """
-
-        dbcollection = self.db[collection_name]
-        result = dbcollection.insert_many(document_dictionaries)
-
-        return result
-
-    def update(self, collection_name, query, update_dictionary,
-               selection, upsert=False, sort=None,
-               return_document=ReturnDocument.BEFORE):
-        """ Finds a document and modify it
+    def write(self, collection_name,
+                    query,
+                    dictionary,
+                    selection,
+                    atomic=False,
+                    upsert=False,
+                    sort=None,
```

Comment:
Forget about last comment. Based on the discussion here #10 it seems atomic operations on many rows is something we should forget about.

Then I think the behaviors for atomic is True or False are too different to be crammed inside the write() method. Since the default behavior is *not* atomic then we could postpone the addition of atomic operation to write/remove and finish this PR with only `write(collection_name, query, dictionary, selection)` (and maybe `upsert` if we decide to make it definable). We can decide later if we integrate atomicity inside write() to keep only write/read/remove or if we add new explicit methods for atomic operations.